server/app.ts

```diff
@@ -34,10 +42,21 @@ export default function createApp(services: Services): express.Application {
   app.use(setUpWebRequestParsing())
   app.use(setUpStaticResources())
   nunjucksSetup(app, path)
-  app.use(setUpAuthentication())
-  app.use(authorisationMiddleware())
+
+  app.use(
+    setUpAuthentication(),
+    rateLimiter({ windowMs: authenticationRateLimitWindow, maxRequests: authenticationRateLimitMaxRequests }),
+  )
+  app.use(
+    authorisationMiddleware(),
+    rateLimiter({ windowMs: authorisationRateLimitWindow, maxRequests: authorisationRateLimitMaxRequests }),
```

Comment:
as you're setting it at the application layer you would only need to add it once but it would affect all route/middleware after you've hooked it in. 

The examples in the doc are adding different rate limiters to different paths - do you just want to do it on the auth ones?

Reply:
My concern (and the issue that was raised by CodeQL) was that the calls to `isAuthenticated` in the authorisationMiddleware are potentially expensive, so need some degree of rate limiting, hence why the authorisation rate limits are more permissive.

Reply:
I don't think this is how [app.use](https://expressjs.com/en/4x/api.html#app.use) works. 

you have said

```
app.use(middleware1, middleware2)
app.use(middleware3, middleware4)
```

which is going to mean every requests will hit the middleware in the order:

middleware1 -> middleware2 -> middleware3 -> middleware4

so you will have no protection on middleware1 (`setUpAuthentication`) and then every request will go through 2 separate middleware requests.

Reply:
the examples in the express-rate-limiter docs show adding rate limiter to specific paths

Reply:
Ah ha, I think I've got it now 👍